### PR TITLE
Drop support for ActiveModel 3.2

### LINKED ISSUE #345: Drop support for ActiveModel 3.2
Like #343, we decided in #338 to drop support for ActiveModel 3.2.

### PR DESCRIPTION
**Pull request by tfausak**
Fixes #345.

### CODE REVIEW

**Comment by tfausak** [comment]
I would like to replace [`ActiveInteraction::Errors::Details`](https://github.com/orgsync/active_interaction/blob/57807f27f00614468c209aff4bbc8c7e8a261ba8/lib/active_interaction/backports.rb#L20-L65) with [`active_model-errors_details`](https://github.com/cowbell/active_model-errors_details), but that gem doesn't work with Rails 5. I want to add something like this to our gem spec:

``` rb
if spec.version_of 'active_model' < 5
  spec.add_dependency 'active_model-errors_details', '~> 1.3'
end
```

But I don't know how to do that. I don't even know if it's possible.

**Comment by AaronLasseigne** [comment]
I have no idea how to do that.

**Comment by tfausak** [comment]
If I had to guess, I'd say it's not possible. I poked around a little in `Gem::Specification` and `Gem::Dependency` and didn't find anything helpful. 

I'm ok with leaving `ActiveInteraction::Errors::Details` in there. We can drop it when we drop support for Rails < 5 :wink:

File: active_interaction.gemspec [hunk 1]
```diff
@@ -31,10 +31,10 @@ Gem::Specification.new do |gem|
     Dir.glob(File.join('lib', 'active_interaction', 'locale', '*.yml'))
   gem.test_files = Dir.glob(File.join('spec', '**', '*.rb'))
 
-  gem.add_dependency 'activemodel', '>= 3.2', '< 6'
+  gem.add_dependency 'activemodel', '>= 4', '< 6'
 
   {
-    'actionpack' => ['>= 3.2', '< 6'],
+    'actionpack' => [],
```
Comment (AaronLasseigne) [inline active_interaction.gemspec]: Good idea. No need to repeat the version numbers.

**Comment by AaronLasseigne** [comment]
:shipit:

**Comment by AaronLasseigne** [comment]
It looks like we can also drop the `HashFilter` code in the backport. It's Rails < 4.0.0.

**Pull request merged**